sale_purchase_force_vendor/models/sale_order_line.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    vendor_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Vendor",
+    )
+    allowed_vendor_ids = fields.One2many(
+        comodel_name="res.partner",
+        string="Proveedor",
+        compute="_compute_allowed_vendor_ids",
+    )
+
+    @api.depends("product_id")
+    def _compute_allowed_vendor_ids(self):
+        supplierinfo_model = self.env["product.supplierinfo"]
+        self.allowed_vendor_ids = self.allowed_vendor_ids
+        for item in self.filtered(lambda x: x.product_id):
```

Comment:
Make this configurable through ICP or similar for restricting or not the allowed vendors for covering the both cases we have.

Reply:
Ok, I have added a checkbox in Purchase > Configuration > Products to restrict or not.